src/provider/util/mod.rs

```diff
@@ -1,3 +1,97 @@
-/// Utilities for provider module
+//! Utilities for provider module.
 pub(in crate::provider) mod fb_msm;
 pub(in crate::provider) mod msm;
+
+#[cfg(test)]
+pub mod test_utils {
+  //! Contains utilities for testing and benchmarking.
+  use crate::spartan::polys::multilinear::MultilinearPolynomial;
+  use crate::traits::{
+    commitment::CommitmentEngineTrait, evaluation::EvaluationEngineTrait, Engine,
+  };
+
+  /// Generates a random polynomial and point from a seed to test a proving/verifying flow of one
+  /// of our EvaluationEngine over a given Engine.
+  pub(crate) fn prove_verify_from_ell<E: Engine, EE: EvaluationEngineTrait<E>>(ell: usize) {
```

Comment:
This function looks great, though I'm not sure about its name: ell isn't a standard nomenclature.
Here are things that would help me grok what this function is for out of context, and that you could use to edit the comment, the name of this function, or more likely both :
- make it clear we're testing the prove and verify flow of *Multilinear* Polynomial Commitment Schemes (PCS), 
- which in Nova are represented by `EvaluationEngineTrait` implementations for proving, 
- replacing `ell` by the more evocative `num_var`, which becomes clear once the first bullet point has been stated clearly,
- avoid the mention of an `Engine`, which is implied and selected by a choice of `EvaluationEngineTrait` implementation,